tests/integration/test_tls.py

```diff
@@ -0,0 +1,383 @@
+
+    assert (
+        ops_test.model.applications[charm_versions.integration_hub.application_name].status
+        == "active"
+    )
+
+
+@pytest.mark.abort_on_fail
+async def test_jdbc_endpoint_with_default_metastore(ops_test: OpsTest, test_pod):
+    """Test the JDBC endpoint exposed by the charm."""
+    logger.info("Running action 'get-jdbc-endpoint' on kyuubi-k8s unit...")
+    kyuubi_unit = ops_test.model.applications[APP_NAME].units[0]
+    action = await kyuubi_unit.run_action(
+        action_name="get-jdbc-endpoint",
+    )
+    result = await action.wait()
+
+    jdbc_endpoint = result.results.get("endpoint")
+    logger.info(f"JDBC endpoint: {jdbc_endpoint}")
+
+    logger.info(
+        "Testing JDBC endpoint by connecting with beeline" " and executing a few SQL queries..."
+    )
+    process = subprocess.run(
+        [
+            "./tests/integration/test_jdbc_endpoint.sh",
+            test_pod,
+            ops_test.model_name,
+            jdbc_endpoint,
+            "db_default_metastore",
+            "table_default_metastore",
+        ],
+        capture_output=True,
+    )
+    print("========== test_jdbc_endpoint.sh STDOUT =================")
+    print(process.stdout.decode())
+    print("========== test_jdbc_endpoint.sh STDERR =================")
+    print(process.stderr.decode())
+    logger.info(f"JDBC endpoint test returned with status {process.returncode}")
+    assert process.returncode == 0
+
+
+@pytest.mark.abort_on_fail
+async def test_enable_ssl(ops_test: OpsTest, charm_versions, test_pod):
+    await asyncio.gather(
+        ops_test.model.deploy(
+            charm_versions.tls.application_name,
+            application_name=charm_versions.tls.application_name,
+            channel=charm_versions.tls.channel,
+            num_units=1,
+            config={"ca-common-name": "kyuubi"},
+            series=charm_versions.tls.series,
+            # FIXME (certs): Unpin the revision once the charm is fixed
+            revision=charm_versions.tls.revision,
+        ),
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, charm_versions.tls.name], status="active", timeout=1000, idle_period=30
+    )
+    await ops_test.model.add_relation(APP_NAME, charm_versions.tls.name)
+
+    async with ops_test.fast_forward(fast_interval="60s"):
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME, charm_versions.tls.name],
+            status="active",
+            timeout=1000,
+            idle_period=30,
+        )
+
+    host = await get_address(ops_test, unit_name=f"{APP_NAME}/0")
+
+    response = subprocess.check_output(
+        f"openssl s_client -showcerts -connect {host}:10009 < /dev/null",
+        stderr=subprocess.PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+
+    assert "TLSv1.3" in response
+    assert "CN = kyuubi" in response
+
+    # get issued certificates
```

Comment:
**out-of-scope,todo** eventually when we will have a proper relation for retrieving credentials/endpoints, say `kyuubi <> data-integrator` we should probably add the certificate/CA via the relation data, and fetch it using the `data-integrator` action. 

To not forget this, maybe can we add a reminder to this with a code comment with a TODO and/or also adding the link of this comment to the ticket that we have about adding the kyuubi client relation? I believe it is [DPE-3942](https://warthogs.atlassian.net/browse/DPE-3942)